brillig_vm/src/black_box.rs

```diff
@@ -0,0 +1,211 @@
+                generic_hash_256::<Sha256>(message, output, registers, memory);
+            }
+            BlackBoxOp::Blake2s { message, output } => {
+                generic_hash_256::<Blake2s256>(message, output, registers, memory);
+            }
+            BlackBoxOp::Keccak256 { message, output } => {
+                generic_hash_256::<Keccak256>(message, output, registers, memory);
+            }
+            BlackBoxOp::HashToField128Security { message, output } => {
+                generic_hash_to_field::<Blake2s256>(message, output, registers, memory);
+            }
+            BlackBoxOp::EcdsaSecp256k1 {
+                hashed_msg,
+                public_key_x,
+                public_key_y,
+                signature,
+                result: result_register,
+            } => {
+                let message_values = memory.read_slice(
+                    registers.get(hashed_msg.pointer).to_usize(),
+                    registers.get(hashed_msg.size).to_usize(),
+                );
+                let message_bytes = to_u8_vec(message_values);
+
+                let public_key_x_bytes: [u8; 32] =
+                    to_u8_vec(memory.read_slice(
+                        registers.get(public_key_x.pointer).to_usize(),
+                        public_key_x.size,
+                    ))
+                    .try_into()
+                    .expect("Expected a 32-element public key x array");
+
+                let public_key_y_bytes: [u8; 32] =
+                    to_u8_vec(memory.read_slice(
+                        registers.get(public_key_y.pointer).to_usize(),
+                        public_key_y.size,
+                    ))
+                    .try_into()
+                    .expect("Expected a 32-element public key y array");
+
+                let signature_bytes: [u8; 64] = to_u8_vec(
+                    memory.read_slice(registers.get(signature.pointer).to_usize(), signature.size),
+                )
+                .try_into()
+                .expect("Expected a 64-element signature array");
+
+                let result = verify_secp256k1_ecdsa_signature(
+                    &message_bytes,
+                    &public_key_x_bytes,
+                    &public_key_y_bytes,
+                    &signature_bytes,
+                );
+
+                registers.set(*result_register, (result as u128).into())
+            }
+        }
+    }
+}
+
+/// Extracts the last byte of every value
+fn to_u8_vec(inputs: &[Value]) -> Vec<u8> {
+    let mut result = Vec::with_capacity(inputs.len());
+    for input in inputs {
+        let field_bytes = input.to_field().to_be_bytes();
+        let byte = field_bytes.last().unwrap();
+        result.push(*byte);
+    }
+    result
+}
+
+/// Does a generic hash of the inputs storing the resulting 32 bytes as items in the output array.
+fn generic_hash_256<D: Digest>(
+    message: &HeapVector,
+    output: &HeapArray,
+    registers: &mut Registers,
```

Comment:
```suggestion
    registers: &Registers,
```

We're not mutating any of the registers here.